apps/publisher/extensions/assets/default/apis/assets.jag

```diff
@@ -35,7 +35,58 @@
  /apis/assets/{id}
 
  */
-
+var assetApi = require('/modules/asset-api.js').api;
+var lcApi = require('/modules/lifecycle/lifecycle-api.js').api;
+var responseProcessor = require('utils').response;
+var constants = require('rxt').constants;
+var addAsset = function (options, req, res, session) {
+    var requestURI = req.getRequestURI();
+    var response;
+    var newAsset = assetApi.create(options, req, res, session);
+    if (!newAsset) {
+        response = responseProcessor.buildErrorResponseDefault(constants.STATUS_CODES.INTERNAL_SERVER_ERROR, 'error on create asset', res, 'Failed to create asset of type: ' + options.type, 'more information on create asset error', []);
+        return;
+    }
+    response = responseProcessor.buildSuccessResponseDefault(constants.STATUS_CODES.CREATED, res, newAsset);
+    response.addHeader('Location', requestURI + '/' + newAsset.id + '?type=' + newAsset.type);
+};
+var updateAsset = function (options, req, res, session) {
+    var response;
+    var updatedAsset = assetApi.update(options, req, res, session);
+    log.info(stringify(updatedAsset));
+    if (!updatedAsset) {
+        response = responseProcessor.buildErrorResponseDefault(constants.STATUS_CODES.NOT_FOUND, 'error on update api', res, 'No matching asset found by id : ' + options.id, 'this is more information', []);
+        return;
+    }
+    response = responseProcessor.buildSuccessResponseDefault(constants.STATUS_CODES.ACCEPTED, res, updatedAsset);
+};
+var getAsset = function (options, req, res, session) {
+    var asset = assetApi.get(options, req, res, session);
+    var response;
```

Comment:
response has been defined as a local variable and that is not returned either. So, we should be able to get rid of that in all the places.